packages/cf-component-icon/src/Icon.js

```diff
@@ -1,32 +1,178 @@
+  door: { content: "'\\e906'" },
+  download: { content: "'\\e900'" },
+  drive: { content: "'\\f00d'" },
+  facebook: { content: "'\\f01f'" },
+  file: { content: "'\\f013'" },
+  filter: { content: "'\\f00e'" },
+  firebolt: { content: "'\\e903'" },
+  flowchart: { content: "'\\f00f'" },
+  gear: { content: "'\\f01c'" },
+  googleplus: { content: "'\\f021'" },
+  hamburger: { content: "'\\f010'" },
+  happy: { content: "'\\f016'" },
+  help: { content: "'\\f01d'" },
+  linkedin: { content: "'\\f022'" },
+  list: { content: "'\\f012'" },
+  lock: { content: "'\\f011'" },
+  mail: { content: "'\\e904'" },
+  network: { content: "'\\e905'" },
+  ok: { content: "'\\f009'" },
+  pause: { content: "'\\f007'" },
+  plus: { content: "'\\f015'" },
+  refresh: { content: "'\\e901'" },
+  remove: { content: "'\\f01a'" },
+  reorder: { content: "'\\f024'", transform: 'rotate(90deg) translateX(-3px)' },
+  sad: { content: "'\\f017'" },
+  search: { content: "'\\f018'" },
+  shield: { content: "'\\f014'" },
+  speech: { content: "'\\f01b'" },
+  time: { content: "'\\f008'" },
+  twitter: { content: "'\\f020'" },
+  upload: { content: "'\\f023'" },
+  widen: { content: "'\\f024'" },
+  wrench: { content: "'\\f019'" }
+};
+
+const Wrapper = createComponent(
+  ({ size, spin = false, type, theme }) => {
+    let iconData = iconMapping[type];
+
+    if (size) {
+      let fontSize = '1em';
 
-    if (this.props.size) {
-      className += ` cf-icon--${this.props.size}`;
+      switch (size) {
+        case '1.5x':
+          fontSize = '1.5em';
+          break;
+        case '2x':
+        case 'large':
+          fontSize = '2em';
+          break;
+        case '2.5x':
+          fontSize = '2.5em';
+          break;
+        case '3x':
+        case 'xlarge':
+          fontSize = '3em';
+          break;
+        case '3.5x':
+          fontSize = '3.5em';
+          break;
+        case '4x':
+        case 'huge':
+          fontSize = '4em';
+          break;
+      }
+      iconData = Object.assign({}, iconData, { fontSize });
     }
 
-    if (this.props.border) className += ' cf-icon--border';
-    if (this.props.spin) className += ' cf-icon--spin';
-    if (this.props.muted) className += ' cf-icon--muted';
-    if (this.props.white) className += ' cf-icon--white';
+    if (spin) {
+      iconData = Object.assign({}, iconData, {
+        display: 'inline-block',
+        animationName: {
+          from: { transform: 'rotate(0deg)' },
+          to: { transform: 'rotate(359deg)' }
+        },
+        animationDuration: '2s',
+        animationIterationCount: 'infinite',
+        animationTimingFunction: 'linear'
+      });
+    }
+
+    const { content, ...rest } = iconData;
+
+    return {
+      color: theme.color,
+      fontStyle: theme.fontStyle,
+      fontSize: theme.fontSize,
+      ...rest,
+      fontFace: {
+        fontFamily: 'Cloudflare Icons',
+        src: [`data:application/x-font-ttf;base64,${cloudflareIcons}`]
```

Comment:
more general question: are we going to deprecate Cloudlfare icons font instead of svg? We brought this question earlier but I don't remember the final decision we made. https://css-tricks.com/icon-fonts-vs-svg/